nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/ui/flow-analysis-rules/flow-analysis-rule-table/flow-analysis-rule-table.component.ts

```diff
@@ -0,0 +1,261 @@
+    }
+
+    getValidationErrorsTipData(entity: FlowAnalysisRuleEntity): ValidationErrorsTipInput {
+        return {
+            isValidating: entity.status.validationStatus === 'VALIDATING',
+            validationErrors: entity.component.validationErrors
+        };
+    }
+
+    hasBulletins(entity: FlowAnalysisRuleEntity): boolean {
+        return !this.nifiCommon.isEmpty(entity.bulletins);
+    }
+
+    getBulletinsTipData(entity: FlowAnalysisRuleEntity): BulletinsTipInput {
+        return {
+            bulletins: entity.bulletins
+        };
+    }
+
+    getStateIcon(entity: FlowAnalysisRuleEntity): string {
+        if (entity.status.validationStatus === 'VALIDATING') {
+            return 'validating fa fa-spin fa-circle-o-notch';
+        } else if (entity.status.validationStatus === 'INVALID') {
+            return 'invalid fa fa-warning';
+        } else {
+            if (entity.status.runStatus === 'DISABLED') {
+                return 'disabled icon icon-enable-false';
+            } else if (entity.status.runStatus === 'DISABLING') {
+                return 'disabled icon icon-enable-false';
+            } else if (entity.status.runStatus === 'ENABLED') {
+                return 'enabled fa fa-flash';
+            } else if (entity.status.runStatus === 'ENABLING') {
+                return 'enabled fa fa-flash';
+            }
+        }
+        return '';
+    }
+
+    formatState(entity: FlowAnalysisRuleEntity): string {
+        if (entity.status.validationStatus === 'VALIDATING') {
+            return 'Validating';
+        } else if (entity.status.validationStatus === 'INVALID') {
+            return 'Invalid';
+        } else {
+            if (entity.status.runStatus === 'DISABLED') {
+                return 'Disabled';
+            } else if (entity.status.runStatus === 'DISABLING') {
+                return 'Disabling';
+            } else if (entity.status.runStatus === 'ENABLED') {
+                return 'Enabled';
+            } else if (entity.status.runStatus === 'ENABLING') {
+                return 'Enabling';
+            }
+        }
+        return '';
+    }
+
+    formatType(entity: FlowAnalysisRuleEntity): string {
+        return this.nifiCommon.formatType(entity.component);
+    }
+
+    formatBundle(entity: FlowAnalysisRuleEntity): string {
+        return this.nifiCommon.formatBundle(entity.component.bundle);
+    }
+
+    isDisabled(entity: FlowAnalysisRuleEntity): boolean {
+        return entity.status.runStatus === 'DISABLED';
+    }
+
+    isEnabledOrEnabling(entity: FlowAnalysisRuleEntity): boolean {
+        return entity.status.runStatus === 'ENABLED' || entity.status.runStatus === 'ENABLING';
+    }
+
+    hasActiveThreads(entity: ReportingTaskEntity): boolean {
+        return entity.status?.activeThreadCount > 0;
+    }
+
+    canConfigure(entity: FlowAnalysisRuleEntity): boolean {
+        return this.canRead(entity) && this.canWrite(entity);
```

Comment:
@mcgilman here also it seems in current NiFi the user can configure a rule even when disabled. Is this correct?